firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -119,7 +134,111 @@ public void onFailure(@NonNull @NotNull Exception e) {
    */
   @NonNull
   public Task<AppDistributionRelease> checkForUpdate() {
-    return Tasks.forResult(null);
+
+    if (checkForUpdateTaskCompletionSource != null
+        && !checkForUpdateTaskCompletionSource.getTask().isComplete()) {
+      checkForUpdateCancellationSource.cancel();
+    }
+
+    checkForUpdateCancellationSource = new CancellationTokenSource();
+    checkForUpdateTaskCompletionSource =
+        new TaskCompletionSource<>(checkForUpdateCancellationSource.getToken());
```

Comment:
Does checkForUpdateHandler need to be a class variable? Feels like it can be moved to the place where its actually being used and also create it in the function

Reply:
@andrewbibiloni did you resolve this by mistake or did you mean to resolve it?

Reply:
Sorry! Didn't refresh the page for changes. Resolving it again